lib/browser.ts

```diff
@@ -624,116 +612,91 @@ export class ProtractorBrowser extends AbstractExtendedWebDriver {
-                    `${timeout}. This may be because the current page is not an Angular ` +
-                    `application. Please see the FAQ for more details: ` +
-                    `https://github.com/angular/protractor/blob/master/docs/timeouts.md#waiting-for-angular`;
-                if (description.indexOf(' - Locator: ') == 0) {
-                  errMsg += '\nWhile waiting for element with locator' + description;
-                }
-                let pendingTimeoutsPromise: wdpromise.Promise<any>;
-                if (this.trackOutstandingTimeouts_) {
-                  pendingTimeoutsPromise = this.executeScriptWithDescription(
-                      'return window.NG_PENDING_TIMEOUTS',
-                      'Protractor.waitForAngular() - getting pending timeouts' + description);
-                } else {
-                  pendingTimeoutsPromise = wdpromise.when({});
-                }
-                let pendingHttpsPromise = this.executeScriptWithDescription(
-                    clientSideScripts.getPendingHttpRequests,
-                    'Protractor.waitForAngular() - getting pending https' + description,
-                    this.internalRootEl);
-
-                return wdpromise.all([pendingTimeoutsPromise, pendingHttpsPromise])
-                    .then(
-                        (arr: any[]) => {
-                          let pendingTimeouts = arr[0] || [];
-                          let pendingHttps = arr[1] || [];
-
-                          let key: string, pendingTasks: string[] = [];
-                          for (key in pendingTimeouts) {
-                            if (pendingTimeouts.hasOwnProperty(key)) {
-                              pendingTasks.push(' - $timeout: ' + pendingTimeouts[key]);
-                            }
-                          }
-                          for (key in pendingHttps) {
-                            pendingTasks.push(' - $http: ' + pendingHttps[key].url);
-                          }
-                          if (pendingTasks.length) {
-                            errMsg += '. \nThe following tasks were pending:\n';
-                            errMsg += pendingTasks.join('\n');
-                          }
-                          err.message = errMsg;
-                          throw err;
-                        },
-                        () => {
-                          err.message = errMsg;
-                          throw err;
-                        });
-              } else {
-                throw err;
-              }
-            });
+        }
+        for (key in pendingHttps) {
+          pendingTasks.push(' - $http: ' + pendingHttps[key].url);
+        }
+        if (pendingTasks.length) {
+          errMsg += '. \nThe following tasks were pending:\n';
+          errMsg += pendingTasks.join('\n');
+        }
+        err.message = errMsg;
+        throw err;
+      } else {
```

Comment:
We always `throw err;` whether or not `if (timeout)` is true, so just extract that from the if statement and drop the `else`